### ISSUE TITLE
Is it possible to pre-cache all the required arguments (names, shapes, etc)?

### ISSUE THREAD

**Issue by aldanor** [nbigaouette/onnxruntime-rs#39]
Why do this every time `run()` is called, would it be possible to do it upon load?

https://github.com/nbigaouette/onnxruntime-rs/blob/32e9935a8b30cffda3a92f458fb7f2dbe35f7412/onnxruntime/src/session.rs#L383-L406

**Comment by nbigaouette** [comment]
Yes, that makes a lot of sense. That came from the original port of the C example. It should definitely be moved.
